01-messaging-crypto-and-init.md

```diff
@@ -590,21 +578,25 @@ creating a decryption oracle.
 
 The structure of transport messages resembles the following:
 ```
-+------------------------------
-|2-byte encrypted packet length|
-+------------------------------
-| 16-byte MAC of the encrypted |
-|        packet length         |
-+------------------------------
-|                              |
-|                              |
-|          ciphertext          |
-|                              |
-|                              |
-+------------------------------
++----------------------------------
+|2-byte encrypted ciphertext length|
++----------------------------------
+|   16-byte MAC of the encrypted   |
+|        ciphertext length         |
++----------------------------------
+|                                  |
+|                                  |
+|             ciphertext           |
+|                                  |
+|                                  |
++----------------------------------
+|         16-byte MAC of the       |
+|            ciphertext            |
++----------------------------------
 ```
 
-The prefixed packet lengths are encoded as a `2-byte` big-endian integer.
+The prefixed packet lengths are encoded as a `2-byte` big-endian integer,
```

Comment:
The rationale for having the packet length size be _outside_ of the protocol message framing is that with this, the encrypted transport is completely decoupled from the protocol message framing. 

Currently, the protocol message themselves are _encapsulated_ within the encrypted transport. Furthermore, we can change the messaging framing arbitrarily without having to modify the encrypted transport messaging. 

Also other chains/coins can freely re-use the encrypted message transport without having to make any additional modifications outside of possibly modifying the inter-chain protocol messages.